frontend/src/test/java/com/google/tripmeout/frontend/servlet/TripParentServletTest.java

```diff
@@ -0,0 +1,133 @@
+    when(storage.getAllUserTrips(TripParentServlet.STUB_USER_ID)).thenReturn(ImmutableList.of());
+
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+
+    tripParentServlet.doGet(request, response);
+
+    assertThat(response.getContentType()).isEqualTo("application/json");
+    assertThat(response.getResponseString()).isEqualTo("[]");
+  }
+  @Test
+  public void doGet_someTrips() throws Exception {
+    // TODO(issues/66): Use specif userId string
+    TripModel trip1 = createTripWithUserId(TripParentServlet.STUB_USER_ID);
+    TripModel trip2 = createTripWithUserId(TripParentServlet.STUB_USER_ID);
+    when(storage.getAllUserTrips(TripParentServlet.STUB_USER_ID))
+        .thenReturn(ImmutableList.of(trip1, trip2));
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    tripParentServlet.doGet(request, response);
+    assertThat(response.getContentType()).isEqualTo("application/json");
+    List<TripModel> tripResponse =
+        gson.fromJson(response.getResponseString(), new TypeToken<List<TripModel>>() {}.getType());
+    assertThat(tripResponse).containsExactly(trip1, trip2);
+  }
+
+  @Test
+  public void doPost_ioExceptionWhenReadingRequest_internalServerErrorResponse() throws Exception {
+    when(request.getReader()).thenThrow(new IOException("something bad happened"));
+    when(request.getContentType()).thenReturn("application/json;");
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    tripParentServlet.doPost(request, response);
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+  }
+  @Test
+  public void doPost_invalidTrip_badRequestResponse() throws Exception {
+    when(request.getReader())
+        .thenReturn(new BufferedReader(new StringReader("not really even json?")));
+    when(request.getContentType()).thenReturn("application/json;");
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    tripParentServlet.doPost(request, response);
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void doPost_wellFormedRequest_fillsInDefaultsAndPersistsTrip() throws Exception {
+    // TODO(issues/66): Use specif userId string
+    TripModel trip = createTripForPost();
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+
+    when(request.getReader()).thenReturn(new BufferedReader(new StringReader(gson.toJson(trip))));
+    tripParentServlet.doPost(request, response);
+    TripModel responseTrip = gson.fromJson(response.getResponseString(), TripModel.class);
+    assertThat(responseTrip.name()).isEqualTo(trip.name());
+    assertThat(responseTrip.locationLat()).isEqualTo(trip.locationLat());
+    assertThat(responseTrip.locationLong()).isEqualTo(trip.locationLong());
+    assertThat(responseTrip.id()).isNotEmpty();
+    assertThat(responseTrip.userId()).isEqualTo(TripParentServlet.STUB_USER_ID);
```

Comment:
nit: perhaps you could make trip have userId of STUB_USER_ID, then you cab just do assertThat(responseTrip).isEqualTo(trip) instead of doing an assertion for each field

Reply:
I think the issue with that is the id's will be different since in my doPost it sets a random UUID

Reply:
oh i see, never mind then